packages/editor/src/store/reducer.js

```diff
@@ -211,6 +236,35 @@ const withInnerBlocksRemoveCascade = ( reducer ) => ( state, action ) => {
 	return reducer( state, action );
 };
 
+/**
+ * Higher-order reducer which targets the combined blocks reducer and handles
+ * the `RESET_BLOCKS` action. When dispatched, this action will replace all
+ * blocks that exist in the post, leaving blocks that exist only in state (e.g.
+ * reusable blocks) alone.
+ *
+ * @param {Function} reducer Original reducer function.
+ *
+ * @return {Function} Enhanced reducer function.
+ */
+const withBlockReset = ( reducer ) => ( state, action ) => {
+	if ( state && action.type === 'RESET_BLOCKS' ) {
+		const visibleClientIds = getNestedBlockClientIds( state.order );
+		return {
```

Comment:
Yeah OK. I've pushed up 7d3d116df which changes things so that we instead augment `RESET_BLOCKS` with a list of client IDs that are referenced by reusable blocks and thus shouldn't be touched.